### ISSUE TITLE
Nested DS models cause stack blow out

### ISSUE THREAD

**Issue by sfanetti** [poteto/ember-changeset#239]
We have a model with a reference to another model that it has a "belongsTo" relationship with.  The retrieval of allKeys before the save does a deep scan of the containing object tree - which has references that are circular.  This blows the stack.  

It would be preferable to have some depth limit in place that can be set or have some means of specifying the relevant properties that constitute a valid change for a model.

**Comment by nucleartide** [comment]
@sfanetti Thanks for the report. This sounds like the same issue mentioned in https://github.com/poteto/ember-changeset/pull/206/commits/68636f23111bea4ed81758730d994904212fad6a.

For now, I'd recommend pinning ember-changeset to 1.3.0 and ember-changeset-validations to 1.2.8. Let me know if that works for you!

**Comment by sdhull** [comment]
FWIW, we encountered this same bug but instead of a stackoverflow, the recursion ended at an error `object.hasOwnProperty is not a function` that was getting swallowed somehow and the `catch` method wasn't firing so it was simply failing silently.

Would recommend you push a fix for this ASAP. Seems like `Changeset` is incompatible with any model that has a `belongsTo`. So our experience seems to contradict [@poteto's assertion](https://github.com/poteto/ember-changeset/issues/118#issuecomment-254302025) that belongsTo should work.

Last thing: We tried manually applying the fix from 68636f23111bea4ed81758730d994904212fad6a (which, btw, where does that commit even exist? It appears to not be on master or any other branch?), however our changeset failed to set the associated model. So while it prevents the error, it still doesn't work with `belongsTo` associations.
